File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Product extends Model
{
    protected $fillable = [
        'company_id',
        'name',
        'sku',
        'brand',
        'strength',
        'pack_size',
        'category',
        'mrp',
        'unit_price',
        'description',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'mrp' => 'decimal:2',
            'unit_price' => 'decimal:2',
            'is_active' => 'boolean',
        ];
    }

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }

    public function orderItems(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    public function formattedPrice(): string
    {
        return \App\Support\Currency::format($this->unit_price);
    }

    public function displayLabel(): string
    {
        $parts = array_filter([$this->name, $this->strength, $this->pack_size]);

        return implode(' · ', $parts);
    }
}

File: mobile_app/lib/models/product.dart
class Product {
  Product({
    required this.id,
    required this.name,
    required this.unitPrice,
    this.sku,
    this.brand,
    this.strength,
    this.packSize,
    this.category,
    this.mrp,
    this.description,
    this.displayLabel,
  });

  final int id;
  final String name;
  final double unitPrice;
  final String? sku;
  final String? brand;
  final String? strength;
  final String? packSize;
  final String? category;
  final double? mrp;
  final String? description;
  final String? displayLabel;

  String get label => displayLabel ?? name;

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as int,
      name: json['name'] as String,
      unitPrice: (json['unit_price'] as num).toDouble(),
      sku: json['sku'] as String?,
      brand: json['brand'] as String?,
      strength: json['strength'] as String?,
      packSize: json['pack_size'] as String?,
      category: json['category'] as String?,
      mrp: json['mrp'] != null ? (json['mrp'] as num).toDouble() : null,
      description: json['description'] as String?,
      displayLabel: json['display_label'] as String?,
    );
  }
}
